core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -154,32 +180,40 @@ object ClientQuotaManager {
  * @param config @ClientQuotaManagerConfig quota configs
  * @param metrics @Metrics Metrics instance
  * @param quotaType Quota type of this quota manager
+ * @param quotaEnforcementType Quota enforcement type of this quota manager. Refers to the
+ *                             documentation of {recordAndGetThrottleTimeMs} for details
  * @param time @Time object to use
+ * @param threadNamePrefix The thread prefix to use
+ * @param clientQuotaCallback An optional @ClientQuotaCallback
  */
 class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
                          private val metrics: Metrics,
                          private val quotaType: QuotaType,
+                         private val quotaEnforcementType: QuotaEnforcementType,
```

Comment:
I have refactored the `ClientQuotaManager` to be a bit more generic and to accept a quota enforcement type.

Reply:
It looks like we don't need this change since the only one that sets a different type does its own thing.

Reply:
Yes, you are right. I was trying to make `ClientQuotaManager` a generic base class for all client quotas. I will still try to extract the real common parts in a base class once but that will be for a follow-up PR.